internal/support/scheduler/init.go

```diff
@@ -0,0 +1,265 @@
+	dec.Prefix = internal.ConfigV2Stem + internal.SupportSchedulerServiceKey
+	dec.ErrCh = errCh
+	dec.UpdateCh = updateCh
+
+	defer dec.Close()
+	defer close(updateCh)
+	defer close(errCh)
+	go dec.Run()
+
+	select {
+	case <-time.After(2 * time.Second):
+		err = errors.New("timeout loading config from registry")
+	case ex := <-errCh:
+		err = errors.New(ex.Error())
+	case raw := <-updateCh:
+		actual, ok := raw.(*ConfigurationStruct)
+		if !ok {
+			return conf, errors.New("type check failed")
+		}
+		conf = actual
+	}
+
+	return conf, err
+}
+
+func listenForConfigChanges() {
+	errCh := make(chan error)
+	dec := consulclient.NewConsulDecoder(Configuration.Registry)
+	dec.Target = &ConfigurationStruct{}
+	dec.Prefix = internal.ConfigV2Stem + internal.SupportSchedulerServiceKey
+	dec.ErrCh = errCh
+	dec.UpdateCh = chConfig
+
+	defer dec.Close()
+	defer close(errCh)
+
+	go dec.Run()
+	for {
+		select {
+		case ex := <-errCh:
+			LoggingClient.Error(ex.Error())
+		case raw, ok := <-chConfig:
+			if ok {
+				actual, ok := raw.(*ConfigurationStruct)
+				if !ok {
+					LoggingClient.Error("listenForConfigChanges() type check failed")
+				}
+				Configuration = actual //Mutex needed?
+			} else {
+				return
+			}
+		}
+	}
+}
+
+func initializeClients(useConsul bool) {
+	// Create metadata clients
+	params := types.EndpointParams{
+		ServiceKey:  internal.SupportSchedulerServiceKey,
+		Path:        clients.ApiDeviceRoute,
+		UseRegistry: useConsul,
+		Url:         Configuration.Clients["Metadata"].Url() + clients.ApiDeviceRoute,
+		Interval:    Configuration.Service.ClientMonitor,
+	}
+
+	mdc = metadata.NewDeviceClient(params, startup.Endpoint{})
+	params.Path = clients.ApiCommandRoute
+	params.Url = Configuration.Clients["Metadata"].Url() + clients.ApiCommandRoute
+	cc = metadata.NewCommandClient(params, startup.Endpoint{})
+}
+
+func setLoggingTarget() string {
+	if Configuration.Logging.EnableRemote {
+		return Configuration.Clients["Logging"].Url() + clients.ApiLoggingRoute
+	}
+	return Configuration.Logging.File
+}
+
+func AddDefaultSchedulers() error {
+
+	LoggingClient.Info(fmt.Sprintf("loading default schedules and schedule events..."))
+
+	defaultSchedules := Configuration.Schedules
+
+	for i := range defaultSchedules {
+
+		defaultSchedule := models.Schedule{
+			BaseObject: models.BaseObject{},
+			Id:         bson.NewObjectId(),
+			Name:       defaultSchedules[i].Name,
+			Start:      defaultSchedules[i].Start,
+			End:        defaultSchedules[i].End,
+			Frequency:  defaultSchedules[i].Frequency,
+			Cron:       defaultSchedules[i].Cron,
+			RunOnce:    defaultSchedules[i].RunOnce,
+		}
+		err := addSchedule(defaultSchedule)
+		if err != nil {
+			return LoggingClient.Error("AddDefaultSchedulers() - failed to load schedule %s", err.Error())
+		} else {
+			LoggingClient.Info(fmt.Sprintf("added default schedule %s", defaultSchedule.Name))
+		}
+	}
+
+	defaultScheduleEvents := Configuration.ScheduleEvents
```

Comment:
Suggesting calling `defaultScheduleEvents` just `events`

Reply:
removed the references to the prefix 'default' from scheduleEvents.  events might be too generic in this case. scheduleEvents is contextual to the use of events.